client/src/components/IntegerInput.tsx

```diff
@@ -0,0 +1,184 @@
+import { Button, InputGroup } from "@blueprintjs/core"
+import React, { useEffect, useRef, useState } from "react"
+import utils from "utils"
+
+const ALLOWED_KEYS = [
+  "Backspace",
+  "Delete",
+  "ArrowLeft",
+  "ArrowRight",
+  "Tab",
+  "Home",
+  "End"
+]
+
+interface IntegerInputProps {
+  min?: number
+  max?: number
+  value?: number
+  onValueChange?: (value: number | undefined) => void
+  placeholder?: string
+}
+
+export default function IntegerInput({
+  min,
+  max,
+  value,
+  onValueChange,
+  placeholder
+}: IntegerInputProps) {
+  const [internalValue, setInternalValue] = useState(undefined)
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const clampValue = val => {
+    if (!utils.isNumeric(val)) {
+      return undefined
+    }
+    let clamped = val
+    if (min !== undefined && clamped < min) {
```

Comment:
Can be changed to:
```javascript
    if (min != null && clamped < min) {
```
(which will check for both `null` and `undefined` since we're using the _equality_ operator instead of _identity_)